tests/unit/utils/test_git_shell.py

```diff
@@ -375,3 +376,31 @@ def test_tar_from_ref_and_filepaths(tmp_path):
         filenames = tar.getnames()
         # THEN only first file in in tar
         assert filenames == [first_file_name]
+
+
+def test_git_ls_unstaged(tmp_path):
+    # GIVEN a repository
+    repo = Repository.create(tmp_path)
+    repo.create_commit()
+
+    # AND a submodule
+    submodule = Repository.create(repo.path / "submodule")
+    submodule.create_commit()
+
+    repo.git("submodule", "add", submodule.path)
+
+    # AND unstaged files
+    repo_file = repo.path / "repo_file"
+    submodule_file = repo.path / "submodule" / "submodule_file"
+    repo_file.write_text("Content")
+    submodule_file.write_text("Content")
+
+    # WHEN listing unstaged files
+    unstaged_files = git_ls_unstaged(wd=repo.path)
+
+    # THEN unstaged files in repo and submodule are returned
+    # as relative OR absolute paths
+    assert all(
+        file.name in [Path(path).name for path in unstaged_files]
+        for file in [repo_file, submodule_file]
+    ), "Not all paths are found in unstaged_files"
```

Comment:
This only tests the file name, not the file path. But `git_ls_unstaged()` implementation is good now. I rewrote this part a bit and it still passes.

```suggestion
    # THEN unstaged files in repo and submodule are returned
    # as relative to repo.path
    expected_paths = {x.relative_to(repo.path) for x in (repo_file, submodule_file)}
    assert {Path(x) for x in unstaged_files} == expected_paths
```

Reply:
`relative_to()` would have saved me some time, trying to work with `\` in path :laughing: